src/util/saveArticles.ts

```diff
@@ -517,7 +525,12 @@ async function treatImage(mw: MediaWiki, dump: Dump, srcCache: KVS<boolean>, art
         }
 
         /* Change image source attribute to point to the local image */
-        img.setAttribute('src', newSrc);
+        if (downloader.webp && isImageUrl(newSrc)) {
+            img.setAttribute('src', newSrc + '.webp');
+        }
+        else {
+            img.setAttribute('src', newSrc);
+        }
```

Comment:
Use a ternary operator. Concerning `isImageUrl` (which I have asked to delete), you will have to create a new method called `isWebpImageUrl()` and please implement it like `isImageUrl()` ie. with a regular expression.

Reply:
I will use a regular expression here but other than that it should be called isImageUrl only i guess because `newSrc` won't be having .webp in it(as it has been taken out of html content directly) we have to concatenate webp in it.

Reply:
shouldConvertImageUrlToWebp() ?

Reply:
ok